pkg/engine/context/resolvers/informerCache.go

```diff
@@ -0,0 +1,92 @@
+package resolvers
+
+import (
+	"context"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+func getSelectorForCache() (labels.Selector, error) {
+	expectedLabels := "cache.kyverno.io/enabled"
+	selector := labels.Everything()
+	requirement, err := labels.NewRequirement(expectedLabels, selection.Exists, nil)
+	if err == nil {
+		selector = selector.Add(*requirement)
+	}
+	return selector, err
+}
+
+func GetResourceInformer(kubeClient kubernetes.Interface, resyncPeriod time.Duration) (kubeinformers.SharedInformerFactory, corev1listers.ConfigMapLister) {
```

Comment:
This package should not be responsible for creating the factory IMHO.

Reply:
Or at least not both the factory and the lister. Maybe the factory is ok.

Reply:
returning factory from here.